File: examples/spring-boot-java/src/main/java/com/example/ChatController.java
package com.example;

import org.jspecify.annotations.NonNull;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api")
record ChatController(
    @NonNull AIService aiService
) {

    @PostMapping("/chat")
    public Mono<ResponseEntity<ChatResponse>> chat(@RequestBody ChatRequest request) {
        return aiService.generateResponse(request.message())
            .map(
                response -> ResponseEntity.ok(new ChatResponse(response))
            )
            .onErrorReturn(ResponseEntity
                .status(HttpStatus.INTERNAL_SERVER_ERROR)
                .body(new ChatResponse("Error processing request"))
            );
    }

    public record ChatRequest(String message) {
    }

    public record ChatResponse(String response) {
    }
}

File: examples/spring-boot-kotlin/src/main/kotlin/com/example/agent/controller/ChatController.kt
package com.example.agent.controller

import com.example.agent.service.AgentService
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.server.ResponseStatusException

private val logger = KotlinLogging.logger {}

@RestController
class ChatController(val agentService: AgentService) {

    @PostMapping(value = ["/chat"])
    suspend fun chat(@RequestBody request: ChatRequest): ChatResponse {
        try {
            val result = agentService.createAndRunAgent(request.prompt)
            return ChatResponse(result)
        } catch (e: Exception) {
            logger.error(e) { "Failed to run an agent" }
            throw ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Failed to run an agent")
        }
    }
}


data class ChatRequest(val prompt: String)

data class ChatResponse(val response: String)
